lang/types/value_test.go

```diff
@@ -1008,6 +1008,40 @@ func TestValueInto0(t *testing.T) {
 		})
 	}
 }
+
+func TestValueInto1(t *testing.T) {
+	// Ensure Into() clears existing elements out of the slice
+	goList := []string{"", "", "three"}
+	valList := &ListValue{
+		T: NewType("[]str"),
+		V: []Value{&StrValue{V: "one"}, &StrValue{V: "two"}},
+	}
+	err := Into(valList, reflect.ValueOf(&goList))
+	if err != nil {
+		t.Errorf("Into(%+v, %+v) failed: %s", valList, goList, err)
+	}
+	if !reflect.DeepEqual(goList, []string{"one", "two"}) {
+		t.Errorf("Into(%+v, %+v) did not clear existing values from the list", valList, goList)
+	}
+
+	// Ensure Into() clears all existing elements out of the map
+	goMap := map[string]string{"3": "three"}
+	valMap := &MapValue{
+		T: NewType("map{str: str}"),
+		V: map[Value]Value{
+			&StrValue{V: "1"}: &StrValue{V: "one"},
+			&StrValue{V: "2"}: &StrValue{V: "two"},
+		},
+	}
+	err = Into(valMap, reflect.ValueOf(&goMap))
+	if err != nil {
+		t.Errorf("Into(%+v, %+v) failed: %s", valMap, goMap, err)
+	}
+	if !reflect.DeepEqual(goMap, map[string]string{"1": "one", "2": "two"}) {
+		t.Errorf("Into(%+v, %+v) did not clear existing values from the map", valMap, goMap)
```

Comment:
I think the error messages aren't supposed to be capitalized... So `func Into...` instead maybe?